### PR TITLE
tools shuffle-files-cleaner

### PR DESCRIPTION
**Pull request by zwangsheng**
### _Why are the changes needed?_
Add Kyuubi tools cache-file-cleaner on kubernetes.
This tools help to start daemonSet on Kubernetes to clean shuffle file and spark cache file.
By config some envs, cache-file-cleaner will help to check some old file(which depending on configuration) and delete them for free dist.
And after first cleaning, if the free capacity lower than threshold, cache file cleaner will start deep clean to help free dist.
With the cache-file-cleaner, add Dockerfile, entrypoint and daemonSet.yaml to help user to use this tool.
Using threadPool to help delete shuffle files.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: tools/kubernetes/docker/shuffle-files-cleaner/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM openjdk:8-slim
```
Comment (yaooqinn) [inline tools/kubernetes/docker/shuffle-files-cleaner/Dockerfile:1]: license

File: tools/kubernetes/docker/shuffle-files-cleaner/cleaner.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+apiVersion: apps/v1
```
Comment (yaooqinn) [inline tools/kubernetes/docker/shuffle-files-cleaner/cleaner.yml:1]: ditto

File: tools/kubernetes/docker/shuffle-files-cleaner/cleaner.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      containers:
+        - image: <image>
+          name: cleaner
+          volumeMounts:
+            - name: spark
+              mountPath: /data
+            - name: cleaner-log
+              mountPath: /log/cleanerLog
+          env:
+            - name: cacheDirs
```
Comment (yaooqinn) [inline tools/kubernetes/docker/shuffle-files-cleaner/cleaner.yml:24]: can we add some comment here for these

File: tools/kubernetes/docker/shuffle-files-cleaner/cleaner.yml (hunk 2 above)
Comment (yaooqinn) [reply tools/kubernetes/docker/shuffle-files-cleaner/cleaner.yml:17]: We need about 200 ~ 500 words explanation here

File: tools/kubernetes/docker/shuffle-files-cleaner/cleaner.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: kyuubi-kubernetes-shuffle-file-cleaner
+  # daemonSet nameSpace
```
Comment (yaooqinn) [inline tools/kubernetes/docker/shuffle-files-cleaner/cleaner.yml:21]: this is not an eligible comment.

File: tools/kubernetes/docker/shuffle-files-cleaner/cleaner.yml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  selector:
+    matchLabels:
+      name: <name>
+  template:
+    metadata:
+      labels:
+        name: <name>
+    spec:
+      containers:
+        # container image
```
Comment (yaooqinn) [inline tools/kubernetes/docker/shuffle-files-cleaner/cleaner.yml:33]: this is not an eligible comment.

File: tools/kubernetes/docker/shuffle-files-cleaner/cleaner.yml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        - image: <image>
+          name: cleaner
+          # volume dirs where store log and need to clean shuffle files
+          volumeMounts:
+            - name: spark
```
Comment (yaooqinn) [inline tools/kubernetes/docker/shuffle-files-cleaner/cleaner.yml:38]: why the name here is `spark`?

File: tools/kubernetes/docker/shuffle-files-cleaner/cleaner.yml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: kyuubi-kubernetes-shuffle-file-cleaner
+  # daemonSet nameSpace
+  namespace: spark
```
Comment (yaooqinn) [inline tools/kubernetes/docker/shuffle-files-cleaner/cleaner.yml:22]: why the namespace is default and default to spark

**Comment by zwangsheng** [comment]
close this PR for creating new one
KYUUBI #768

**Pull request closed without merging**